crypto/evp/keymgmt_meth.c

```diff
@@ -154,10 +155,20 @@ static void *keymgmt_from_algorithm(int name_id,
             break;
         case OSSL_FUNC_KEYMGMT_IMPORT_TYPES:
             if (keymgmt->import_types == NULL) {
-                importfncnt++;
+                if (importtypesfncnt == 0)
```

Comment:
Might be simpler just to increment here?
and then just add && !=3 to the test on line 212

Reply:
I wanted to avoid that some own can supply just the two `import_types` functions but not the import function. That would result in `importfncnt = 2` and this would be let through in the check below. 
So the check really must be that the import function and at least one of the import-types functions (or both) is specified. That's the reason for the somewhat more complicated logic.